### ISSUE TITLE
proposal:add new option to rule `array-callback-return` and `getter-return`

### ISSUE THREAD

**Issue by Aladdin-ADD** [eslint/eslint#8539]
**What rule do you want to change?**
array-callback-return

**Does this change cause the rule to produce more or fewer warnings?**
fewer.

**How will the change be implemented? (New option, new default behavior, etc.)?**
new option: 
"allowImplicit": true -- allows implicitly returning undefined with a `return;` statement.

**Please provide some example code that this change will affect:**

```js
/* eslint array-callback-return: [2, {allowImplicit: true} */
[1, 2, 3].map(function (item) {
  return; // return undefined implicitly.
});
```

**What does the rule currently do for this code?**
Function expected a return value.(array-callback-return)

**What will the rule do after it's changed?**
no error.

**Comment by ljharb** [comment]
Yes please - I never write `return undefined;` because `undefined` is not a keyword, and is not a safe token to rely on in non-app code.

I would love this option (and would enable it in the airbnb config).

**Comment by eslintbot** [comment]
Hi @Aladdin-ADD, thanks for the issue. It looks like there's not enough information for us to know how to help you.

If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

Requesting a rule change? Please see [Proposing a Rule Change](http://eslint.org/docs/developer-guide/contributing/rule-changes) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by platinumazure** [comment]
Hi @Aladdin-ADD, are you thinking we should enhance the rule with an option like for getter-return? Or should we just focus on cleaning up the documentation?

**Comment by ljharb** [comment]
I would like an option.

**Comment by Aladdin-ADD** [comment]
the rule getter-return's option `allowImplicit` has been accepted. so better to be consistent.

**Comment by platinumazure** [comment]
@ljharb You could use `return void 0;`, that's perfectly safe.

That said, I'm :+1: to there being an option.

**Comment by platinumazure** [comment]
@eslint/eslint-team A PR just came for this, we should probably evaluate quickly or else close.

**Comment by Aladdin-ADD** [comment]
@platinumazure @gyandeeps  updated the OP.

**Comment by ilyavolodin** [comment]
@eslint/eslint-team Does anyone else want's to support this change?

**Comment by kaicataldo** [comment]
> "allowImplicit": true -- disallows implicitly returning undefined with a return; statement.

**Comment by Aladdin-ADD** [comment]
eh.yes! corrected @kaicataldo

**Comment by ilyavolodin** [comment]
Marking as accepted since @Aladdin-ADD is part of the team and will be the champion.
